mapping/src/main/scala/nl/lumc/sasc/biopet/pipelines/mapping/MultisampleMappingReport.scala

```diff
@@ -114,10 +114,29 @@ trait MultisampleMappingReportTrait extends MultisampleReportBuilder {
   }
 
   /** Files page, can be used general or at sample level */
-  def filesPage(sampleId: Option[Int] = None, libraryId: Option[Int] = None): ReportPage = {
-    val flexiprepExecuted = Await.result(summary.getStatsSize(runId, "flexiprep", NoModule, mustHaveLibrary = true), Duration.Inf) >= 1
-
-    ReportPage(List(), Nil, Map())
+  def filesPage(sampleId: Option[Int] = None, libraryId: Option[Int] = None): Future[ReportPage] = {
```

Comment:
In this one function I see several await calls even though the return type is already `Future`. Is it not possible to use `Future.flatMap` or `Future.map` instead? So you can do `Await` on whoever calls `filesPage` instead?